_docs/tasks/security/https-overlay.md

```diff
@@ -0,0 +1,214 @@
+When Istio sidecar is deployed with an https service, the proxy automatically downgrades
+from L7 to L4 (no matter mTLS is enabled or not), which means it does not terminate the
+original https traffic. And this is the reason Istio can work on https services.
+
+## Before you begin
+
+* Set up Istio by following the instructions in the
+  [quick start]({{home}}/docs/setup/kubernetes/quick-start.html).
+  Note that authentication should be **disabled** at step 5 in the
+  [installation steps]({{home}}/docs/setup/kubernetes/quick-start.html#installation-steps).
+
+## Deploy an https service without Istio sidecar
+
+### Generate certificates
+
+You need to have openssl installed to run this command
+
+```bash
+$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/nginx.key -out /tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"
+```
+
+```bash
+$ kubectl create secret tls nginxsecret --key /tmp/nginx.key --cert /tmp/nginx.crt
+secret "nginxsecret" created
+```
+
+### Create an https service without Istio sidecar
+
+This subsection creates a nginx-based https service.
+
+First we need to create a configmap:
+
+```bash
+$ kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf
+configmap "nginxconfigmap" created
+```
+
+Then deploy the service:
+
+```bash
+$ kubectl apply -f samples/https/nginx-app.yaml
+...
+service "nginxsvc" created
+replicationcontroller "my-nginx" created
+```
+
+Then, create another pod to call this service.
+
+```bash
+$ kubectl apply -f <(bin/istioctl kube-inject --debug -f samples/sleep/sleep.yaml)
+```
+
+Get the pods
+
+```bash
+$ kubectl get pod
+NAME                              READY     STATUS    RESTARTS   AGE
+my-nginx-jwwck                    2/2       Running   0          1h
+sleep-847544bbfc-d27jg            2/2       Running   0          18h
+```
+
+Ssh into the istio-proxy container of sleep pod.
+```bash
+$ kubectl exec -it sleep-847544bbfc-d27jg -c istio-proxy /bin/bash
+```
+
+Call nginxsvc
+```bash
+# curl https://nginxsvc -k
+...
+<h1>Welcome to nginx!</h1>
+...
+```
+
+You can actually combine the above two command into one:
+
+```bash
+$ kubectl exec sleep-847544bbfc-d27jg -c istio-proxy -- curl https://nginxsvc -k
+...
+<h1>Welcome to nginx!</h1>
+...
+```
+
+### Create an https service with Istio sidecar with mTLS disabled
+
+In "Before you begin" section, the istio control plane is deployed with mTLS
+disabled. So you only need to redeploy the nginx https service with sidecar.
+
+Delete the https service.
+```bash
+$ kubectl delete -f nginx-app.yaml
+```
+
+Deploy it with sidecar
+
+```bash
+$ kubectl apply -f <(bin/istioctl kube-inject --debug -f samples/https/nginx-app.yaml)
+```
+
+Make sure the pod is up and running
+
+```bash
+$ kubectl get pod
+NAME                              READY     STATUS    RESTARTS   AGE
+my-nginx-6svcc                    2/2       Running   0          1h
+sleep-847544bbfc-d27jg            2/2       Running   0          18h
+```
+
+And run
```

Comment:
Similar, change to single command to make it easy to follow.

Reply:
I do have the one line command listed as an alternative. Some users (including me) may feel one line is harder to debug (e.g., you can't ssh into the pod).